docs/DeveloperGuide.md

```diff
@@ -183,11 +196,80 @@ detailed view:
   * Pros: Easy to keep UI up to sync with model by overriding ListCell's updateItem method
   * Cons: Can allow for displaying of multiple DetailedBookCards even though the detailed view is currently only meant to show
   one book
-  
+
 * **Alternative 2:** Use other JavaFX layouts
   * Pros: More in-line with the purpose of the detailed view of showing only one book
   * Cons: More work has to be done to sync up the UI with the model.
 
+
+### Add book
+
+####Implementation
+
+*bookmark* allows Users to add books into the application.
+
+This feature is faciltated mainly by `LogicManager`, `AddCommandParser` and `AddCommand`.
+
+![Classes involved in the Add Command](images/LogicClassDiagram.png)
+
+`LogicManager#execute()` handles the command word to create `AddCommandParser` to parse the remaining inputs.
+`AddCommandParser#parse()` tokenizes each prefix to create a `Book` object. This `Book` object will be
+passed as a parameter to create a `AddCommand` that will be returned to `LogicManager`<br>
+*If there are missing or invalid prefixes, an exception will be thrown with a message to the User.*
+
+`LogicManager#execute()` will call `AddCommand#execute()` to add the `Book` attribute into
+the `Model`to return a `CommandResult` as a feedback to the user.
+*If there is an existing book with the same name, an exception will be thrown with a message to the User*
+
+Below is an activity diagram which illustrates the flow of events for adding a book
+
+![Add command flow of execution](images/AddActivityDiagram.png)
+
+Below is a sequence diagram which illustrates the a scenario where a User adds a valid book <br>
+Command: `add n/Harry Potter g/Fiction tp/1000 b/100`
+
+![Interactions inside the logic component for the add command](images/AddSequenceDiagram.png)
+
+
+### Suggestion feature
+
+#### Implementation
+
+*bookmark*'s Did you mean? feature uses the Damerau-Levenshtien algorithm to calculate the distance between the
+user-input word and the words in the application Library.
+
+This feature is facilitated by mainly by `SuggestionAlgorithm` and `WordBank` and has coupling with
+`ModelManager` and `FindCommand`.
+
+The class diagram below shows the relevant classes involved.
+
+![Suggestion Algorithm and the Classes involved](images/AlgorithmClassDiagram.png)
+
+Given below is an example usage scenario and how a Suggestion mechanism behaves at each step.
+
+Step 1: The user inputs the command `find n/h@rry` to find books with `harry` in their names. <br>
+*harry is deliberately mispelled* <br>
+`FindCommand` will implement `execute` and the `model`'s `FilteredList`  will be empty.
+
+Step 2: `FindCommand` will call on `SuggestionAlgorithm#findSuggestions()` to find the closest matching word
+in the appropriate `WordStore` of `WordBank`.
+
+Step 3: `SuggestionAlgorithm#calculateDistance()` will be implemented to calculate the edit distance of `h@rry` and the words in `nameWordBank`
```

Comment:
Did you mean "will be called" rather than "will be implemented"?

Reply:
Thanks for pointing this out! I have made the changes.